protecode/scanning_util.py

```diff
@@ -377,41 +387,45 @@ def enumerate_reference_triages():
                 self._api.rescan(protecode_app.product_id())
 
         # upload new images
-        for resource in images_to_upload:
+        for upload_element in elements_to_upload:
             try:
                 scan_result = self._upload_image(
-                    component=resource_group.component(),
-                    resource=resource,
+                    component=upload_element.component,
+                    resource=upload_element.resource,
                 )
+
             except requests.exceptions.HTTPError as e:
                 # in case the image is currently being scanned, Protecode will answer with HTTP
                 # code 409 ('conflict'). In this case, fetch the ongoing scan to add it
                 # to the list of scans to consider. In all other cases re-raise the error.
                 if e.response.status_code != requests.codes.conflict:
                     raise e
 
-                image_ref = resource.access.imageReference
+                image_ref = upload_element.resource.access.imageReference
                 logger.warning(f'conflict whilst trying to upload {image_ref=}')
 
                 scan_result = self.retrieve_scan_result(
-                    component=resource_group.component(),
-                    resource=resource,
+                    component=upload_element.component,
+                    resource=upload_element.resource,
                 )
 
+            # map created products
+            product_id_to_upload_element[scan_result.product_id()] = upload_element
             protecode_apps_to_consider.append(scan_result)
 
-        # wait for all apps currently being scanned
-        for protecode_app in protecode_apps_to_consider:
-            # replace - potentially incomplete - scan result
-            protecode_apps_to_consider.remove(protecode_app)
-            logger.info(f'waiting for {protecode_app.product_id()}')
-            protecode_apps_to_consider.append(
-                self._api.wait_for_scan_result(protecode_app.product_id())
-            )
-            logger.info(f'finished waiting for {protecode_app.product_id()}')
+        def wait_for_scan_results(protecode_apps_to_consider):
+            for protecode_app in protecode_apps_to_consider:
+                logger.info(f'waiting for {protecode_app.product_id()}')
+                yield self._api.wait_for_scan_result(protecode_app.product_id())
+                logger.info(f'finished waiting for {protecode_app.product_id()}')
 
+        scan_results = list(
+            wait_for_scan_results(
+                protecode_apps_to_consider=protecode_apps_to_consider,
+            )
+        )
         # apply imported triages for all protecode apps
-        for protecode_app in protecode_apps_to_consider:
+        for protecode_app in scan_results:
```

Comment:
we should rename many variable and refactor some parts to simplify maintenance, but not part of this PR :)

Reply:
+1 - please keep the semantical focus of this PR on corecting the bug, and add a refactoring commit afterwards.